core/chains/evm/bulletprooftxmanager/common.go

```diff
@@ -0,0 +1,61 @@
+package bulletprooftxmanager
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/ethereum/go-ethereum/rpc"
+	"github.com/pkg/errors"
+
+	evmclient "github.com/smartcontractkit/chainlink/core/chains/evm/client"
+	"github.com/smartcontractkit/chainlink/core/logger"
+)
+
+// Tries to send transactions in batches. Even if some batch(es) fail to get sent, it tries all remaining batches,
+// before returning with error for the latest batch send. If a batch send fails, this sets the error on all
+// elements in that batch.
+// TODO: The batch send is just sending the batch to 1 RPC node. Change it to send to all nodes,
+// similar to how Pool.SendTransaction sends a transaction to all nodes.
```

Comment:
Can we share logic with the eth resender, which does in fact do this?

Reply:
Why would we not touch broadcast_at here? Seems like we probably should, no?

Reply:
Ya. I didn't realize what it means and its importance. But it doesn't hurt.
So touching it now. 
The batchSendTransaction is now shared between ethResender and ethConfirmer code.